docs: add clarification for pre-vouching contributors (#11096)
ghostty-org/ghostty: CONTRIBUTING.md

longer trust-by-default because it makes it too trivial to generate
plausible-looking but actually low-quality contributions.

> ## Pre-Vouching Contributors

Reviewer: I think “Pre-Vouching Contributors” sounds like “contributors prior to being vouched” rather than “those who contributed prior to the introduction of the vouch system”. I can't think of a better phrasing though; “Existing Contributors” is too vague, “Contributors From Before the Vouch System” sounds too wordy; maybe “Contributors Preceding the Vouch System” or “Prior To” might work,

> If you have already contributed to Ghostty prior to the introduction

Reviewer 2: ```suggestion
If you contributed to Ghostty prior to the introduction
```

> of the vouch system and wish to continue contributing, you are not

Reviewer 2: ```suggestion
of the vouch system and wish to continue contributing, you were not
```

> automatically vouched. You are required to follow the same

Reviewer 2: ```suggestion
automatically added to the [list of vouched users](.github/VOUCHED.td). You will need to follow the same
```
